app/src/main/java/ru/flicksbox/user/domain/UserInteractor.kt

```diff
@@ -0,0 +1,70 @@
+package ru.flicksbox.user.domain
+
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import ru.flicksbox.App
+import ru.flicksbox.R
+import ru.flicksbox.data.AppError
+import ru.flicksbox.data.Data
+import ru.flicksbox.user.presentation.UserInteractor
+
+const val emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
```

Comment:
please use built in approach to check the email correctness
`val isEmailCorrect = Patterns.EMAIL_ADDRESS.matcher("some@email.com").matches()`